### ISSUE TITLE
Establish licencing and contribution terms for all the projects

### ISSUE THREAD

**Issue by alysbrett** [ScottishCovidResponse/SCRCIssueTracking#60]
I notice that not all the repos have licence files yet and I understand that the RAMPrequirement that they are open source. We should also have contribution guidelines/agreements that cover IP issues. 

@richardreeve says: I'd also quite like to limit the numbers of licenses to maybe GPL 3+ or 2-clause BSD (perhaps) just to reduce confusion, and possibly ask people to agree to submit contributions under the 2-clause BSD in either case so that if we decide to switch one from GPL -> BSD for any reason we don't have to go around seeking approval?

Any thoughts on the above (either comment here or contact Richard and I)

Could owners/lead RSEs tick here when a licence has been agreed and added please? (Also let me know if the abbreviation below is not the best shorthand for each project - would be useful to agree one)

- [ ] BICI

- [ ] TP

- [ ] FMD

- [ ] ND

- [ ] Simulation.jl

- [ ] Simple network Sim

**Comment by bobturneruk** [comment]
We're planning to use 2 clause BSD - is there a preferred statement on copyright holder? There are differences between projects at present.

**Comment by richardreeve** [comment]
No, so far we haven't taken a position. Copyright was already a mess beforehand tbh, so I never suggested anything. Have some projects asked for copyright assignment? I know the julia model just made up a "Scottish COVID-19 Response Consortium" entity, which doesn't really exist because we thought it was easier, but maybe it isn't?

**Comment by bobturneruk** [comment]
We've followed suit as we need a proper license in place.
```
Copyright 2020 Jessica Enright and the Scottish COVID-19 Response Consortium
```
We could change, if advised.

**Comment by bobturneruk** [comment]
I plan to add this for contributions, based on what's there on the Julia model:

## Licensing

By contributing to this project (e.g. by submitting a pull request or providing advice on code), you agree - unless simultaneously and expressly stated otherwise - that your contribution may be included in the source code of the project and published under the [2-Clause BSD License](https://opensource.org/licenses/BSD-2-Clause) and that the contribution was created in whole or in part by you and you have the right to submit it under the open source license indicated above.

**Comment by ianhinder** [comment]
Technically, this probably needs approval of our individual institutions.  By default, University of Manchester owns some stake in the IP that I produce (I don't remember what, exactly - I think they have a nonexclusive use clause).  Normally, my work is internal to UoM, so this doesn't matter.  I don't think, for example, that I have the authority to just "assign" copyright of my work to someone else.  I don't think this is going to be a real problem in practice, but is something we probably need to sort out the paperwork for.

**Comment by bobturneruk** [comment]
Yes - we should not be held up by this, but remain aware of it. I might get some advice, informally. I'll feedback anything of interest.

**Comment by ianhinder** [comment]
I've been advised that while it's possible to do a transfer of copyright to an external entity, setting up such an entity is a lot of work, and the transfer is also quite paperwork-heavy.  So it's much easier to just leave the copyright with the institutions.  That would mean each code having copyright owned by multiple institutions, I suppose.  I don't know if that is a problem (who would decide on things like relicensing?).  One example I found stated something along the lines of "Copyright administered by UoM on behalf of <list of people who contributed to the code>".

**Comment by richardreeve** [comment]
To be honest, I think in general we're happy for whoever wants to add themselves to copyright statements to do so wherever they want to. We're not planning on relicensing anything beyond the statement in `Simulation.jl` that hopefully allows us to relicense the GPL-3+ code as 2-Clause BSD if we decide to down the road.

**Comment by ianhinder** [comment]
I've just checked, and as far as I can see, none of the models mention copyright assignment in their Contributors document.  If someone contributes to a project, they own the copyright to their contribution, unless they have expressly assigned it to someone else.  Many open source  projects require an emailed statement of copyright assignment before a contribution will be accepted.

**Comment by ianhinder** [comment]
There's an interesting page about this issue [here](https://producingoss.com/en/contributor-agreements.html).  The author does *not* recommend actual copyright assignment.   Instead, the contributor agrees that the contribution is licensed under the terms of the project.  This appears to be what users agree to in the GitHub terms of service, and is a widely-understood convention in the open source world.  Having an email from each contributor to this effect wouldn't hurt.  So the end result would be that the code is copyright all the contributors, but is licensed according to the project's existing licence.  You might then be able to say "Copyright The Authors" in the copyright statement, as long as it's clear how to determine all the contributors.

**Comment by richardreeve** [comment]
> This appears to be what users agree to in the GitHub terms of service, and is a widely-understood convention in the open source world. Having an email from each contributor to this effect wouldn't hurt.
That's probably not a bad idea...

**Comment by ianhinder** [comment]
BEEPmbp licence and contributing docs finally in place. That's all the models now!  Note that there are many more repositories containing source code in SCRC than are listed in the description of this issue.
